internal/applicationsnapshot/vsa.go

```diff
@@ -22,35 +22,47 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"time"
 
+	ecc "github.com/enterprise-contract/enterprise-contract-controller/api/v1alpha1"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/afero"
 )
 
-// SnapshotVSAWriter handles writing application snapshot VSA predicates to files
-type SnapshotVSAWriter struct {
+// SnapshotPredicate represents a predicate for an entire application snapshot
+type SnapshotPredicate struct {
```

Comment:
Question, maybe not related to this PR:

Do we have a different predicate for just one component, or is the the predicate type for all VSAs?

Reply:
The predicate type is for all VSAs.